prow/life_of_a_prow_job.md

```diff
@@ -15,7 +19,13 @@ type PluginClient struct {
+"hook" [multiplexes events](https://github.com/kubernetes/test-infra/blob/c8829eef589a044126289cb5b4dc8e85db3ea22f/prow/cmd/hook/server.go#L40) by looking at "X-GitHub-Event", a custom http header. Afterwards, a [PluginAgent object](https://github.com/kubernetes/test-infra/blob/c8829eef589a044126289cb5b4dc8e85db3ea22f/prow/plugins/plugins.go#L86), initialized [during Hook's startup](https://github.com/kubernetes/test-infra/blob/c8829eef589a044126289cb5b4dc8e85db3ea22f/prow/cmd/hook/main.go#L128), selects plugins to handle events. See [events.go](https://github.com/kubernetes/test-infra/blob/c8829eef589a044126289cb5b4dc8e85db3ea22f/prow/cmd/hook/events.go#L17) for more details, and check [plugins.yaml](https://github.com/kubernetes/test-infra/blob/c8829eef589a044126289cb5b4dc8e85db3ea22f/prow/plugins.yaml) for a list of plugins per repo.
+
+"hook" delivers an event that represents my `/test all` comment to the [Trigger plugin](https://github.com/kubernetes/test-infra/tree/c8829eef589a044126289cb5b4dc8e85db3ea22f/prow/plugins/trigger). The trigger plugin uses a function called [`handleGenericCommentEvent` (implementec by `handleGenericComment`)](https://github.com/kubernetes/test-infra/blob/99b91b56b097e39d70cb1ae82c0b1cb57d98ac48/prow/plugins/trigger/generic-comment.go#L32) to process `/test all`.
+
+`handleGenericComment` verifies that it is "ok to test", and then stamps a label that reads "ok to test" on the issue. Finally, `handleGenericComment` determines *presubmit* jobs to run. For `/test all`, it will consider jobs that need no people interaction. The list of jobs supplied by the `Config` object, in the `PluginClient` object, will be used to find suitable jobs.
+
+I requested that some tests run, so the trigger plugin talks to the Kubernetes API server and creates a [ProwJob Custom Resource](https://github.com/kubernetes/test-infra/blob/c8829eef589a044126289cb5b4dc8e85db3ea22f/prow/kube/prowjob.go#L50-L83) with the information from the issue comment. At this point, `cmd/hook`'s role is done. It goes on and handles the next webhooks.
```

Comment:
The whole paragraph started out with `I comment /test all` or not? That means `trigger` will check which presubmits are mandatory for your PR and run those, so the `I requested that some tests run` doesn't sound accurate to me :thinking:

Reply:
fixed :-)